types/ember-data/types/registries/transform.d.ts

```diff
@@ -1,6 +1,8 @@
-export default interface TransformRegistry {
-    string: string;
-    boolean: boolean;
-    number: number;
-    date: Date;
+import DS from 'ember-data';
+
+export default interface TransformRegistry extends Record<string, DS.Transform> {
+    string: DS.StringTransform;
+    boolean: DS.BooleanTransform;
+    number: DS.NumberTransform;
+    date: DS.DateTransform;
```

Comment:
Returning the actual transforms allows us to limit the registry to be transforms only. This avoids an error that we had in our own app where added the return values instead of the transforms themselves.

Reply:
I like this change, but I agree with @chriskrycho that it's too breaking for now.